### PR TITLE
fix: addon to opens-gadget there was still a missing piece. Also adding the bitnami images to Test27 (they generate evidence)

### PR DESCRIPTION
**Pull request by entlein**
https://github.com/k8sstormcenter/node-agent/actions/runs/31890008928/job/95025507994
and the latest one at:
https://github.com/k8sstormcenter/node-agent/actions/runs/31902615394
full details at
https://github.com/k8sstormcenter/node-agent/issues/81

Also opened a draft PR to IG https://github.com/inspektor-gadget/inspektor-gadget/pull/5753

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved filesystem path detection for detached or separately mounted procfs filesystems.
  * Prevented incorrect path construction in buffer-offset and numeric-prefixed path edge cases.

* **Tests**
  * Expanded coverage for procfs alerts, application profiles, learned profiles, and path-handling safeguards.
  * Added validation for a rendered standalone Bitnami Redis deployment, including networking, storage, security, and health checks.

* **Documentation**
  * Clarified the main branch name in the project README.

### CODE REVIEW

File: pkg/ebpf/gadgets/trace_open/filesystem_patched.h [hunk 1]
```diff
@@ -120,7 +124,14 @@ static __always_inline void *get_path_str(struct path *path)
...
 				continue;
 			}
-			// Global root - path fully parsed
+			// Detached procfs mounts (fsopen/fsmount, no mountpoint) canonicalize to /proc
+			if (BPF_CORE_READ(dentry, d_sb, s_magic) == PROC_SUPER_MAGIC) {
+				char proc_name[4] = { 'p', 'r', 'o', 'c' };
+				buf_off -= 1;
+				bpf_probe_read(&(string_p->buf[buf_off & (MAX_PERCPU_BUFSIZE - 1)]), 1, &slash);
+				buf_off -= 4;
+				bpf_probe_read(&(string_p->buf[buf_off & ((MAX_PERCPU_BUFSIZE >> 1) - 1)]), 4, proc_name);
```
[the remark this answers is not in the document]
Comment (entlein) [reply pkg/ebpf/gadgets/trace_open/filesystem_patched.h:133]: Thats correct, needs to be fixde, test added

**Comment by entlein** [comment]
AI: please add the postgres and redis images to Test27, the ones that we found the bug in originally, the nginx (currently in Test27) doesnt seem to be too reliable.

**Comment by entlein** [comment]
Please use the EXACT SAME redis manifest that renders when we use the pinned bitnami installer in the example/redis/distros.sh test

**Comment by entlein** [comment]
AI: there is a manifest still missing, and also     
component_test.go:1493: 
        ========== Test_27 Summary ==========
    component_test.go:1505:   [FAIL] recorded_profile_absolute_paths     profile=(auto-learned)            file=(nginx + bitnami-redis startup) expect NO alert
    component_test.go:1507:          -> found non-absolute or '.' paths in recorded profile 
    
  ->  Compare the fork version and this version of the CT, run it on the PG again .  `run ONLY THE NON CP-deprecated stack`

**Comment by ConstanzeTU** [comment]
The identical Test_27 leg (exact distro-demo render, digest-pinned, with the deterministic gadget positive control) is green in the fork CT harness: https://github.com/k8sstormcenter/node-agent/actions/runs/31902615394 — full iteration ledger on the fork's issue tracker (k8sstormcenter/node-agent#81). The head branch needs the final two leg refinements (probe error surfacing + redis-profile-gated wait) to match; note the DCO check currently fails on the web-edited commits (missing sign-off).

**Comment by matthyx** [comment]
Reviewed the diff and the `get_path_str` change in detail.

## Core fix (`pkg/ebpf/gadgets/trace_open/filesystem_patched.h`)

Traced the `buf_off` arithmetic for the new procfs-prepend block:

- It only fires when `dentry == mnt_root && mnt_p == mnt_parent_p`, i.e. a truly detached mount (no parent to escape into) — a normally-mounted `/proc` still walks up through its real mountpoint via the existing branch above, so there's no double-prepend for ordinary procfs mounts.
- With the `buf_off >= 5` guard, the write sequence (`-1` → slash at full-buffer mask, `-4` → `"proc"` at half-buffer mask) never underflows: worst case `buf_off == 5` lands the `"proc"` write at offset 0. Below 5 the prepend is correctly skipped rather than wrapping.
- The trailing "add leading slash" step in the unmodified tail of the function still runs afterward, so the final string is `/proc/<rest>` with a single separator, not `//proc...`.

This matches the guard the commit message describes and I couldn't find a case where it produces a malformed or out-of-bounds path. The new `traceopen_bufoff_guard_test.go` is a documentation/trip-wire test (doesn't exercise the eBPF code, just pins the arithmetic + regexes the header), but `Test_27`'s new bitnami-redis leg in `component_test.go` does exercise it end-to-end against a real detached-procfs reproducer, which is the part that actually matters here.

## Blocker: DCO check is failing

6 commits are missing `Signed-off-by`:
`0cff334`, `e2826b0`, `17a9acd`, `f41a1e2`, `16be345`, `6e5c40b`.

This needs a signed rebase before merge:
```
git rebase HEAD~9 --signoff
git push --force-with-lease origin upstream-pr/opens-gadget-procfs
```

## Non-blocking notes

- `component-tests` is skipped in this repo's CI (expected for fork-originated PRs — no code issue), so the new `Test_27` leg hasn't actually run here. @ConstanzeTU already reported it green in the fork's own CT harness in this thread, which covers it, but worth a maintainer re-run/confirmation before merging since it's the only real end-to-end validation of the eBPF change.
- `build-and-push-image` failing is also expected/pre-existing behavior for fork PRs in this repo (no registry credentials available to forks — same failure shows up on other recently-merged fork PRs like #889), not something introduced by this change.
- Minor nit in `tests/component_test.go`: `"sigs.k8s.io/yaml"` got added to the stdlib import group instead of the third-party group — cosmetic only.

Logic looks sound to me; holding off on approval only because of the DCO gate. Once the sign-offs are fixed (and ideally the component-tests leg is confirmed green here too), this is good to merge from a code standpoint.

File: README.md [hunk 2]
```diff
@@ -1,3 +1,5 @@
+This fork main = `mirrormain`
+
```
[the remark this answers is not in the document]
Comment (entlein) [reply README.md]: Can you please delete this comment, this is not helpful

**Review by matthyx — APPROVED** [review]
merging once Test_18 rerun turns green
